test_vm/tests/replica_update_test.rs

```diff
@@ -981,6 +982,140 @@ fn deal_included_in_multiple_sectors_failure() {
     v.assert_state_invariants();
 }
 
+#[test]
+fn replica_update_verified_deal() {
+    let store = &MemoryBlockstore::new();
+    let mut v = VM::new_with_singletons(store);
+    let addrs = create_accounts(&v, 3, TokenAmount::from_whole(100_000));
+    let (worker, owner, client, verifier) = (addrs[0], addrs[0], addrs[1], addrs[2]);
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let policy = Policy::default();
+    let (maddr, robust) = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from_whole(10_000),
+    );
+
+    // Get client verified
+    let datacap = StoragePower::from(32_u128 << 30);
+    verifreg_add_verifier(&v, verifier, datacap.clone());
+    verifreg_add_client(&v, verifier, client, datacap);
+
+    // advance to have seal randomness epoch in the past
+    let v = v.with_epoch(200);
+
+    let sector_number = 100;
+    let (v, d_idx, p_idx) = create_sector(v, worker, maddr, sector_number, seal_proof);
+
+    let old_sector_info = sector_info(&v, maddr, sector_number);
+    // make some deals, chop off market's alloc term buffer from deal lifetime.  This way term max can
+    // line up with sector lifetime AND the deal has buffer room to start a bit later while still fitting in the sector
+    let deal_ids = create_verified_deals(
+        1,
+        &v,
+        client,
+        worker,
+        maddr,
+        old_sector_info.expiration - v.get_epoch() - policy.market_default_allocation_term_buffer,
+    );
+
+    // replica update
+    let new_cid = make_sealed_cid(b"replica1");
+    let replica_update = ReplicaUpdate2 {
+        sector_number,
+        deadline: d_idx,
+        partition: p_idx,
+        new_sealed_cid: new_cid,
+        deals: deal_ids.clone(),
+        update_proof_type: fvm_shared::sector::RegisteredUpdateProof::StackedDRG32GiBV1,
+        replica_proof: vec![],
+        new_unsealed_cid: make_piece_cid(b"unsealed from itest vm"),
+    };
+    let updated_sectors: BitField = apply_ok(
+        &v,
+        worker,
+        robust,
+        TokenAmount::zero(),
+        MinerMethod::ProveReplicaUpdates2 as u64,
+        ProveReplicaUpdatesParams2 { updates: vec![replica_update] },
+    )
+    .deserialize()
+    .unwrap();
+    assert_eq!(vec![100], bf_all(updated_sectors));
+
+    // sanity check the sector after update
+    let new_sector_info = sector_info(&v, maddr, sector_number);
+    assert_eq!(1, new_sector_info.deal_ids.len());
+    assert_eq!(deal_ids[0], new_sector_info.deal_ids[0]);
+    assert_eq!(old_sector_info.sealed_cid, new_sector_info.sector_key_cid.unwrap());
+    assert_eq!(new_cid, new_sector_info.sealed_cid);
```

Comment:
Is there an easy way to assert the change in power that should result from this Snap?

Reply:
Maybe not easy but we can get this by asserting all the subcalls which we really should be doing anyway